src/utils/hooks/useLoadReview.tsx

```diff
@@ -1,34 +1,126 @@
 import { useEffect, useState } from 'react'
-import { useGetReviewQuery } from '../generated/graphql'
+import { ApolloError } from '@apollo/client'
+import {
+  ApplicationResponse,
+  ReviewResponse,
+  TemplateElement,
+  useGetReviewQuery,
+} from '../generated/graphql'
+import { ApplicationElementStates, SectionElementStates } from '../types'
+import useLoadApplication from './useLoadApplication'
+import useGetResponsesAndElementState from './useGetResponsesAndElementState'
+import getPageElements from '../helpers/getPageElements'
 
 interface UseLoadReviewProps {
   reviewId: number
+  serialNumber: string
 }
 
-const useLoadReview = ({ reviewId }: UseLoadReviewProps) => {
+const useLoadReview = ({ reviewId, serialNumber }: UseLoadReviewProps) => {
+  const [applicationName, setApplicatioName] = useState<string>('')
+  const [reviewSections, setReviewSections] = useState<SectionElementStates[]>()
   const [loading, setLoading] = useState(true)
   const [error, setError] = useState('')
 
+  const {
+    error: applicationError,
+    application,
+    templateSections,
+    isApplicationLoaded,
+  } = useLoadApplication({
+    serialNumber,
+  })
+
+  const { error: responsesError, responsesByCode, elementsState } = useGetResponsesAndElementState({
+    serialNumber,
+    isApplicationLoaded,
+  })
+
   const { data, error: apolloError, loading: apolloLoading } = useGetReviewQuery({
     variables: {
       reviewId,
     },
+    skip: !isApplicationLoaded || !elementsState,
   })
 
+  const getResponse = (code: string) => {
+    if (responsesByCode && responsesByCode[code]) return responsesByCode[code]
+    else return null
+  }
+
+  useEffect(() => {
+    if (applicationError) {
+      const error = applicationError as ApolloError
+      setError(error.message)
+      return
+    }
+    if (application) setApplicatioName(application.name)
+  }, [applicationError, application])
+
   useEffect(() => {
+    if (responsesError) {
+      setError(responsesError)
+      return
+    }
     if (apolloError) {
       setError(apolloError.message)
       return
     }
-    if (data) {
+    if (data && data.review && data.review.reviewResponses) {
```

Comment:
```suggestion
    if (data?.review?.reviewResponses) {
```
Optional chaining?

Reply:
Yes, I guess that's alright! I was thinking what if it doesn't even have the `data` but that would all resolve to `undefined` you are right.